lib/use-checkout-pricing.js

```diff
@@ -37,33 +44,57 @@ import cloneDeep from 'lodash/cloneDeep';
  */
 export default function useCheckoutPricing(initialInputs = {}, handleError = throwError) {
   const recurly = useRecurly();
-  const [loading, setLoading] = useState(false);
+  const [loading, setLoading] = useState(true);
   const [input, setInput] = useState(initialInputs);
-  let checkoutPricing = recurly.Pricing.Checkout();
-  const [pricing, setPricing] = useState(checkoutPricing);
+  const [pricing, setPricing] = useState(recurly.Pricing.Checkout());
 
   useEffect(() => {
     setLoading(true);
-    const { subscriptions = [], adjustments = [] } = input;
+    const { subscriptions = [], adjustments = [], ...singleInputs } = input;
+    let checkoutPricing = recurly.Pricing.Checkout();
+
+    if (Object.keys(singleInputs).length) {
+      checkoutPricing = addSingleInputs(singleInputs, checkoutPricing);
+    };
 
     if (adjustments.length) {
       checkoutPricing = addAdjustments(adjustments, checkoutPricing);
     };
 
     addSubscriptions(subscriptions, checkoutPricing)
       .then(() => {
-        setPricing(checkoutPricing);
+        checkoutPricing = checkoutPricing.reprice().done(() => {
+          setPricing(checkoutPricing);
+          setLoading(false);
+        });
       })
-      .catch(handleError)
-      .finally(() => setLoading(false));
+      .catch(error => {
+        handleError(error);
+        setLoading(false);
+      });
   }, [input]);
 
   function addAdjustments (adjustments, checkoutPricing) {
     return adjustments
       .reduce((checkoutPricing, adjustment) => {
-        return checkoutPricing.adjustment(adjustment);
-      }, checkoutPricing)
-      .catch(handleError);
+        if (adjustment.itemCode) {
```

Comment:
Without this check, when `adjustment.itemCode` is an empty string, this error is returned: 

![image](https://user-images.githubusercontent.com/32269552/73574360-2dc70b80-443b-11ea-81ba-f3122fb48ca3.png)

Reply:
Mmm interesting. We can look at adding some protection to `CheckoutPricing.adjustment` then.